crypto/fipsmodule/evp/evp.c

```diff
@@ -156,6 +158,76 @@ int EVP_PKEY_cmp(const EVP_PKEY *a, const EVP_PKEY *b) {
   return -2;
 }
 
+char *EVP_get_pw_prompt(void) {
+    return (char*)"Enter PEM passphrase:";
+}
+
+int EVP_read_pw_string(char *buf, int length, const char *prompt, int verify) {
+  return EVP_read_pw_string_min(buf, 0, length, prompt, verify);
+}
+
+int EVP_read_pw_string_min(char *buf, int min_length, int length,
+                           const char *prompt, int verify) {
+  int ret = -1;
+  char verify_buf[1024];
+
+  if (!buf || min_length <= 0 || min_length >= length) {
+    return -1;
+  }
+
+  if (prompt == NULL) {
+    prompt = EVP_get_pw_prompt();
+  }
+
+  // Proactively zeroize |buf| and verify_buf
+  OPENSSL_cleanse(buf, sizeof(buf));
```

Comment:
```suggestion
  OPENSSL_cleanse(buf, length);
```
This should be `length` I think? Otherwise you are just getting the size of the pointer no?